lib/Futures/SharedState.h

```diff
@@ -203,18 +199,14 @@ class SharedState {
             return;
           }
           TRI_ASSERT(state == State::OnlyCallback);  // race with setCallback
-#ifndef _MSC_VER
           [[fallthrough]];
-#endif
 
         case State::OnlyCallback:
-          if (_state.compare_exchange_strong(state, State::Done, std::memory_order_acquire)) {
+          if (_state.compare_exchange_strong(state, State::Done, std::memory_order_release)) {
```

Comment:
Same as above.

Reply:
Should also be fixed via https://github.com/arangodb/arangodb/pull/10291/commits/31044a399c259be5780b467a19dc25dd2b0e7d2f